src/host/os_chmod.c

```diff
@@ -21,7 +21,7 @@ int os_chmod(lua_State* L)
 
 	int mode = (int)strtol(modeStr, &endPtr, 8);
 
-#if PLATFORM_WINDOWS
+#if PLATFORM_WINDOWS && !defined(PLATFORM_CYGWIN)
```

Comment:
If PLATFORM_WINDOWS is defined then PLATFORM_CYGWIN is not; they are mutually exclusive (as they should be) in premake.h.